specs/eventing/data-plane.md

```diff
@@ -1,125 +1,179 @@
-delivery request and respond with an error code if the header is not present.
-
-### Data plane contract for Sources
-
-See [Source Delivery specification](sources.md#source-event-delivery)
-for details.
-
-### Data plane contract for Channels
-
-See [Channel Delivery specification](channel.md#data-plane) for details.
-
-### Data plane contract for Brokers
-
-See [Broker Delivery specification](broker.md)
-
-## Changelog
-
-- 2020-04-20: `0.13.x release`: initial version that documents common contract
-  for sinks, sources, channels and brokers.
+Event senders MAY add or update CloudEvents attributes before sending to
+implement observability features such as tracing; in particular, the
+`traceparent` and `tracestate` distributed tracing attributes defined by
+[W3C](https://www.w3.org/TR/trace-context/) and
+[CloudEvents](https://github.com/cloudevents/spec/blob/v1.0/extensions/distributed-tracing.md)
+may be modified in this way for each delivery attempt of the same event.
+
+This specification does not mandate any particular logging or metrics
+aggregation, nor a method of exposing observability information to users
+configuring the resources. Platform administrators SHOULD expose event-delivery
+telemetry to users through platform-specific interfaces, but such interfaces are
+beyond the scope of this document.
+
+<!-- TODO: should we mention RECOMMENDED spans or RECOMMENDED metrics like in
+https://github.com/knative/specs/blob/main/specs/eventing/channel.md#observability?
+-->
+
+### Derived (Reply) Events
+
+In some applications, an event receiver might emit an event in reaction to a
+received event. Components MAY to choose to support this pattern by accepting an
+encoded CloudEvent in the HTTP response. The sender SHOULD NOT assume that a
+received reply event is directly related to the event sent in the HTTP request.
+
+An event sender MAY document support for this pattern by including a
+`Prefer: reply` header in the HTTP POST request. This header indicates to the
+event receiver that the caller will accept a
+[`200` response](#event-acknowledgement-and-repeat-delivery) which includes a
+CloudEvent encoded using the binary or structured formats.
+[Brokers](./overview.md#broker) and [Channels](./overview.md#channel) MUST
+indicate support for replies using the `Prefer: reply` header.
+
+A recipient MAY reply to any HTTP POST with a `200` response to indicate that
+the event was processed successfully, with or without a response payload. If the
+recipient will _never_ provide a response payload, the `202` response code is
+likely a better choice.
+
+If a recipient chooses to reply to a sender with a `200` response code and a
+reply event in the absence of a `Prefer: reply` header, the sender SHOULD treat
+the event as accepted, and MAY log an error about the unexpected payload. The
+sender MUST NOT process the reply event if it did not advertise the
+`Prefer: reply` capability.
```

Comment:
I'm not sure we can mandate what a sender does w.r.t. the reply at all since we can't verify it. Or put another way, we can only control the interactions between the sender and receiver, not the sender's actions after that interaction.

Likewise, it feels wrong to push people to treat it as an error to get a response w/o the Prefer header. There are going to be lots of sinks that have no idea about this header and to encourage the sender to treat them all as errors doesn't feel right. Especially when they can't actually do anything about it. Granted it's a MAY but it feels like we're encouraging it.